5calls/app/src/main/java/org/a5calls/android/a5calls/controller/SettingsActivity.java

```diff
@@ -113,10 +118,10 @@ public static void updateNotificationsPreference(FiveCallsApplication applicatio
                                                      String result) {
         accountManager.setNotificationPreference(application, result);
         if (TextUtils.equals("0", result)) {
-            OneSignal.disablePush(false);
-            OneSignal.promptForPushNotifications(true);
+            OneSignal.getUser().getPushSubscription().optIn();
```

Comment:
looking at this, we'll probably want to only `optIn` after the permission request is finished being approved -- maybe you could add a TODO for this?